### PR TITLE
Add var and std to DataFrame groupby

### PR DESCRIPTION
**Pull request by mrocklin**
cc @sinhrks

### CODE REVIEW

File: dask/dataframe/groupby.py [hunk 1]
```diff
@@ -68,6 +68,36 @@ def _max(g):
 def _count(g):
     return g.count()
 
+def _var_chunk(df, index):
+    if isinstance(df, pd.Series):
+        df = df.to_frame()
+    x = df.groupby(index).sum()
+    x2 = (df**2).rename(columns=lambda c: c + '-x2')
```
Comment (sinhrks) [inline dask/dataframe/groupby.py]: This squared part is not used in latter part?

Comment (mrocklin) [inline dask/dataframe/groupby.py]: We concat `x2` with `df` in the next line and then use that in our result.

Comment (mrocklin) [inline dask/dataframe/groupby.py]: I may not understand the concern.

File: dask/dataframe/groupby.py [hunk 2]
```diff
@@ -68,6 +68,36 @@ def _max(g):
...
+    return result
+
+def _var_agg(g, ddof):
+    g = g.groupby(level=0).sum()
+    nc = len(g.columns)
+    x = g[g.columns[:nc//3]]
+    x2 = g[g.columns[nc//3:2*nc//3]].rename(columns=lambda c: c[:-3])
+    n = g[g.columns[-nc//3:]].rename(columns=lambda c: c[:-6])
+
+    result = x2 - x**2 / n
```
Comment (sinhrks) [inline dask/dataframe/groupby.py]: I thought either above or here can work.

Comment (mrocklin) [inline dask/dataframe/groupby.py]: I'm still not sure what you mean.  Can you clarify?

Comment (mrocklin) [inline dask/dataframe/groupby.py]: Generally we need to compute the sum of the squares of each numeric column and then group by the index.  We compute the variance through `E(x**2) - E(x)**2`.  This isn't the best approach, but it is the easiest to implement :)

**Comment by mrocklin** [comment]
I'd like to merge soon if possible.

**Comment by mrocklin** [comment]
Merged.  Thanks!

**Pull request merged**